src/clipping/sutherlandhodgman.jl

```diff
@@ -14,6 +14,40 @@ The Sutherland-Hodgman algorithm for clipping polygons.
 """
 struct SutherlandHodgman <: ClippingMethod end
 
-function clip(poly::Polygon, other::Geometry, ::SutherlandHodgman)
-  @error "not implemented"
+function clip(poly::Polygon{Dim,T}, other::Polygon, ::SutherlandHodgman) where {Dim,T}
+  vother = vertices(other)
+  n = length(vother)
+  v = vertices(poly)
+
+  for i in 1:n
+    l₁ = Line(vother[i], vother[mod1(i+1, n)])
+
+    m = length(v)
+    newv = Point{Dim,T}[]
+    
+    for j in 1:m
+      p₁, p₂ = v[j], v[mod1(j+1, m)]
+      l₂ = Line(p₁, p₂)
+
+      # assuming convex clockwise other
```

Comment:
Is it right to assume a clockwise window?

Reply:
All our polygons are oriented counter-clockwise. The holes inside a PolyArea are oriented clockwise.